exercises/triangle/description.md

```diff
@@ -2,6 +2,16 @@ The program should raise an error if the triangle cannot exist.
 
 ## Hint
 
-The sum of the lengths of any two sides of a triangle always exceeds or 
-is equal to the length of the third side, a principle known as the _triangle
+In mathematics, a degenerate case is a limiting case in which an element of a
+class of objects is qualitatively different from the rest of the class and hence
+belongs to another, usually simpler, class.
+
+1. If the sum of the lengths of any two sides of a triangle is equal to the
+   length of the third side, there exists a degenerate triangle.
+
+2. The sum of the lengths of any two sides of a triangle always exceeds or is
```

Comment:
> A corollary to the triangle inequality theorem is there are two classes of triangles--degenerate and nondegenerate.

Thanks, I think this is good. Explicitly noting them will prevent the confusion.

> at least for the Ruby and Coffescript exercises, the second to last test is for the degenerate case
> Interestingly enough I don't see a test for the degenerate case in the Javascript exercise.

Ah. It appears that Javascript did have it at one point, but removed it for https://github.com/exercism/xjavascript/pull/164 .

> If it is a mistake that the Ruby and Coffeescript exercises are testing for the degenerate case, then I'm happy to make pull requests to remove those tests.

Ah, good point. Yes, the test should be removed altogether, as had been decided at https://github.com/exercism/x-common/pull/311#issuecomment-236520834 - tests for degenerate triangles are not in https://github.com/exercism/x-common/blob/master/exercises/triangle/canonical-data.json and tracks should be using this data, unless they have specific language-level concerns (and this likely isn't one).

> If you choose not to test for the degenerate case, then a way to document in the README how this knowledge about degenerate triangles applies to the exercise would be to add a "Dig Deeper" or "Extra Credit" or "Challenge" section where you explain how none of the tests checked for a degenerate triangle, but you can challenge the reader to add those tests. 

I like this! I don't have a strong opinion on which of the three proposed terms to use. If you *made* me pick I'd pick "dig deeper", but I can always be convinced otherwise.

> You can even put the whole explanation of the corollary to the triangle inequality theorem in the "Challenge" section itself

I do think this sounds like a good idea.